src/client/dfuse/il/int_posix.c

```diff
@@ -781,6 +796,12 @@ dfuse_open(const char *pathname, int flags, ...)
 	if (!ioil_iog.iog_initialized || (fd == -1))
 		return fd;
 
+	if (!dfuse_check_valid_path(pathname)) {
+		DFUSE_LOG_DEBUG("open(pathname=%s) ignoring by path",
+				pathname);
+		return fd;
+	}
+
```

Comment:
Why don't we check this before calling `__real_open`?

Reply:
Because this is an interception library.  We want the open call to happen, the intent here is not to prevent files in /proc being opened but to speed up the process.